File: apps/web/src/types/stored_file.ts
export interface StoredFile {
  id: string;
  original_filename: string;
  content_type: string;
  size_bytes: number;
  entity_type: string;
  entity_id: string | null;
  step_index: number | null;
  uploaded_by: string;
  created_at: string;
  url: string | null;
}

File: apps/api/app/schemas/stored_file.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class StoredFileResponse(BaseModel):
    id: uuid.UUID
    original_filename: str
    content_type: str
    size_bytes: int
    entity_type: str
    entity_id: uuid.UUID | None
    step_index: int | None = None
    uploaded_by: uuid.UUID
    created_at: datetime
    url: str | None = None

    model_config = {"from_attributes": True}
